src/kernel/polyhedron.lisp

```diff
@@ -118,6 +118,22 @@
 (defmethod face-points-array ((polyh polyhedron) (face list))
   (coerce (face-points-list polyh face) 'vector))
 
+(defmethod triangles-list ((polyh polyhedron) &key (matrix nil))
+  (let ((triangles '())
+        (tri-polyh (if (is-triangulated-polyhedron? polyh)
+                       polyh
+                       (triangulate-polyhedron polyh))))
+    (flet ((transform-if (xs) (if matrix (transform-points xs matrix) xs)))
+      (do-array (_ face (faces tri-polyh))
+        (push (transform-if (face-points-array tri-polyh face)) triangles)))
+    triangles))
+
+(defmethod triangles-array ((polyh polyhedron) &key (matrix nil))
+  (coerce (triangles-list polyh :matrix matrix) 'vector))
+
+(defmethod triangles-world-array ((polyh polyhedron))
+  (triangles-array polyh :matrix (transform-matrix (transform polyh))))
+
```

Comment:
Please review if these added functions are reasonable, the triangulation is based on existing code in the same file. Regarding naming of the added functions: since there was a `face-points-array` function which gets the face points, a similar nomenclature was used to name this function `triangles-array` which gets an array of triangles representing the polyhedron. Please suggest better names if any.

Reply:
Looks reasonable. Just FYI the current triangulation expects convex polygons only. I don't currently handle concave polygons.

Reply:
Thanks for taking a look. Have added a todo in the [PR description above](https://github.com/kaveh808/kons-9/pull/222#issue-1751565075) regarding the convex polygons precondition. Will address this, maybe in a subsequent PR once the basic object picking is done.

Reply:
> Just FYI the current triangulation expects convex polygons only. I don't currently handle concave polygons.

@kaveh808 I don't have much of a math background but after searching about triangulating concave polyhedrons, it seems non trivial, at least for me:)  However if this could be useful to kons-9 roadmap, I can dig deeper into it. btw I came across [CGAL](https://www.cgal.org/) which might do the triangulation we need. Could CGAL be useful? As you rightly mentioned elsewhere its also important to keep very few dependencies and in which case it might be better to postpone the use of such a library until we really need it at a later stage?